src/jit/codegenarm.cpp

```diff
@@ -152,7 +152,7 @@ void                CodeGen::genSetRegToIcon(regNumber     reg,
     else
     {
         // TODO-CQ: needs all the optimized cases
-        getEmitter()->emitIns_R_I(INS_mov, emitActualTypeSize(type), reg, val);
+        instGen_Set_Reg_To_Imm(emitActualTypeSize(type), reg, val);
```

Comment:
This line subsumes the (val == 0) case also. In general, look at codegenarm64.cpp, and do whatever it has done. In this case, it only calls instGen_Set_Reg_To_Imm() for this function.